src/handle_scope.rs

```diff
@@ -55,3 +70,61 @@ impl<'sc> AsMut<Isolate> for HandleScope<'sc> {
     unsafe { v8__HandleScope__GetIsolate(self) }
   }
 }
+
+#[repr(C)]
+/// A HandleScope which first allocates a handle in the current scope
+/// which will be later filled with the escape value.
+pub struct EscapableHandleScope<'sc>([usize; 4], PhantomData<&'sc mut ()>);
+
+impl<'sc> EscapableHandleScope<'sc> {
+  pub fn new(isolate: &mut impl AsMut<Isolate>) -> Self {
+    let isolate = isolate.as_mut();
+    let mut scope: MaybeUninit<Self> = MaybeUninit::uninit();
+    unsafe {
+      v8__EscapableHandleScope__CONSTRUCT(&mut scope, isolate);
+      scope.assume_init()
+    }
+  }
+
+  /// Pushes the value into the previous scope and returns a handle to it.
+  /// Cannot be called twice.
+  pub fn escape<'parent>(
+    &mut self,
+    mut value: Local<'sc, Value>,
+  ) -> Local<'parent, Value> {
+    unsafe {
+      Local::from_raw(v8__EscapableHandleScope__Escape(self, &mut *value))
+        .unwrap()
+    }
+  }
+}
+
+impl<'sc> Drop for EscapableHandleScope<'sc> {
+  fn drop(&mut self) {
+    unsafe { v8__EscapableHandleScope__DESTRUCT(self) }
+  }
+}
+
+impl<'sc> AsRef<EscapableHandleScope<'sc>> for EscapableHandleScope<'sc> {
+  fn as_ref(&self) -> &Self {
+    self
+  }
+}
```

Comment:
what is this for?

Reply:
Just copy-pasted from `HandleScope` so they have same API